### PR TITLE
feat: ORA autosave enhancements

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Remove "Save Response" button in favor of better autosaving functionality. Paired with better error handling and style cleanup.

JIRA: [AU-1237](https://2u-internal.atlassian.net/browse/AU-1237)

ora-autosave

**What changed?**

- Autosave improvements:
  - Increase autosave cadence to every ~4 seconds (2 second poll + 2 second wait for inactivity).
  - Update autosave text to be shorter and contain icons.
  - Update help text to indicate that responses are autosaved.
  - Remove save button and save button code.

- Error handling:
  - Update autosave logic to be able to continue after an autosave error (e.g. internet dropout).
  - Update error messaging to indicate whether the issue was frontend or backend ("Please check your internet connection" versus "Please contact support staff.")

- General UI cleanup:
  - Remove redundant submission "tip" section.
  - Add line between text and file upload sections.
  - Move "Supported file types" to its own line. Fixes a weird wrapping issue when this is too long.
  - Make padding / margins more consistent throughout.
  - Various SCSS cleanup, removing unused rules, etc.

**Example Screenshot of Team ORA with files & text:**

Screenshot of Team ORA with files and text

**Example screenshot of ORA with files only:**

Screenshot of ORA with files only

---

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled
- [ ] Bumped version number in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3)

**Testing Instructions**

Install locally (since this works now 😉 ) verify the following cases:
- [x] Text responses autosave at the expected cadence.
- [x] Text responses can recover and autosave after an internet drop.

Additionally, verify that the following cases still present correctly:
- [x] Text only.
- [x] Files only.
- [x] Text & files.
- [x] Team responses.
- [x] Multiprompt.

**Browser tests**

Verify that this looks correct on the following browsers:
- [x] Chrome(ium)
- [x] Safari
- [x] Firefox

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

File: openassessment/xblock/submission_mixin.py [hunk 1]
```diff
@@ -258,11 +258,11 @@ def save_submission(self, data, suffix=''):  # pylint: disable=unused-argument
                     {"saved_response": self.saved_response}
                 )
             except Exception:  # pylint: disable=broad-except
-                return {'success': False, 'msg': self._("This response could not be saved.")}
+                return {'success': False, 'msg': self._("Please contact support staff.")}
```
Comment (leangseu-edx) [inline openassessment/xblock/submission_mixin.py:261]:
```suggestion
                return {'success': False, 'msg': self._("Draft could not be saved. Please contact support staff.")}
```

File: openassessment/xblock/static/js/src/oa_server.js [hunk 2]
```diff
@@ -234,7 +234,7 @@ export class Server {
       }).done(function (data) {
         if (data.success) { defer.resolve(); } else { defer.rejectWith(this, [data.msg]); }
       }).fail(function () {
-        defer.rejectWith(this, [gettext('This response could not be saved.')]);
+        defer.rejectWith(this, [gettext('Please check your internet connection.')]);
```
Comment (leangseu-edx) [inline openassessment/xblock/static/js/src/oa_server.js:237]:
```suggestion
        defer.rejectWith(this, [gettext('Draft could not be saved. Please check your internet connection.')]);
```

File: openassessment/xblock/static/js/src/lms/oa_response.js [hunk 3]
```diff
@@ -468,14 +470,22 @@ export class ResponseView {
...
+          this.baseView.toggleActionError('save', null);
+          view.errorOnLastSave = false;
         }
       }).fail((errMsg) => {
-        view.saveStatus(gettext('Error'));
-        view.baseView.toggleActionError('save', errMsg);
+        // Debounce error banner, this won't capture new errors, but will keep
+        // us from defocusing text area, allowing user to continue to edit their
+        // response.
+        if (!view.errorOnLastSave) {
```
[the remark this answers is not in the document]
Comment (nsprenkle) [reply openassessment/xblock/static/js/src/lms/oa_response.js:485]: The comment explains, but running `toggleActionError` defocuses the text field. This causes annoying page refreshes *and* makes it hard to keep typing because the text box loses focus every few seconds.

File: openassessment/xblock/submission_mixin.py (hunk 1 above)
Comment (nsprenkle) [reply openassessment/xblock/submission_mixin.py:261]: The error banner already includes "We could not save your progress" so it seemed unnecessary for it to restate that "Draft could not be saved" as below. I'm not super opinionated about this though, if you think it is important to change.

> **We could not save your progress**
> Draft could not be saved. Please contact support staff.

File: openassessment/xblock/static/js/src/lms/oa_response.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (nsprenkle) [reply openassessment/xblock/static/js/src/lms/oa_response.js:485]: ^^ It will either get stuck in the "Error" or "Draft Saving" state, depending on when the async events fire, but this felt like a small enough issue that I didn't want to sink a bunch of time into fixing it.

File: openassessment/xblock/static/js/src/oa_server.js (hunk 2 above)
Comment (nsprenkle) [reply openassessment/xblock/static/js/src/oa_server.js:237]: See https://github.com/openedx/edx-ora2/pull/1981#discussion_r1244084683

File: openassessment/templates/openassessmentblock/response/oa_response.html [hunk 4]
```diff
@@ -273,10 +272,10 @@ <h5 class="message__title">{% trans "We could not delete files" %}</h5>
                                       {% trans "Select a file to upload for this submission." %}
                                   {% endif %}
                                 </label>
-                                <input type="file" class="submission__answer__upload file--upload" id="submission_answer_upload_{{ xblock_id }}" {% if allow_multiple_files %}multiple=""{% endif %} accept="{{white_listed_file_types|join:", "}}">
-                                <span>
+                                <div>
                                   {% trans "Supported file types: " %}{{ white_listed_file_types|join:", "}}
-                                </span>
+                                </div>
```
Comment (jansenk) [inline openassessment/templates/openassessmentblock/response/oa_response.html:277]: Good call. This looks better

File: openassessment/xblock/static/js/spec/lms/oa_response.js [hunk 5]
```diff
@@ -234,35 +234,31 @@ describe("OpenAssessment.ResponseView", function() {
...
+        expect(view.saveStatus()).toContain('Saving draft');
     });
 
-    it("updates submit/save buttons when response text is optional but file upload is required", function() {
+    it("updates submit button when response text is optional but file upload is required", function() {
         view.textResponse = 'optional';
         view.fileUploadResponse = 'required';
 
         expect(view.submitEnabled()).toBe(false);
-        expect(view.saveEnabled()).toBe(false);
```
Comment (jansenk) [inline openassessment/xblock/static/js/spec/lms/oa_response.js:262]: Should this have
expect(view.saveStatus()).toContain('This response has not been saved');
?

File: openassessment/xblock/static/js/spec/oa_server.js [hunk 6]
```diff
@@ -425,7 +425,7 @@ describe("OpenAssessment.Server", function() {
         stubAjax(false, null);
         var receivedMsg = null;
         server.save("Test").fail(function(errorMsg) { receivedMsg = errorMsg; });
-        expect(receivedMsg).toContain('This response could not be saved');
+        expect(receivedMsg).toContain('Please check your internet connection');
```
Comment (jansenk) [inline openassessment/xblock/static/js/spec/oa_server.js:428]: Maybe concatenate the two messages?

**Review by jansenk — APPROVED** [review]

**Review by leangseu-edx — APPROVED** [review]

File: openassessment/xblock/static/js/spec/lms/oa_response.js (hunk 5 above)
Comment (nsprenkle) [reply openassessment/xblock/static/js/spec/lms/oa_response.js:262]: I hate this checking on strings because it is brittle to any copy changes... but, in lieu of an actual data check, probably?
